src/components/CardList/Card/Card.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { MouseEventHandler, useState } from 'react';
+import s from '../Cards.module.css';
+import { useAppDispatch } from '../../../hooks';
+import { setModalWindow } from '../../../Features/CardsSlice';
+
+export type CardProps = {
+  id: number;
+  name: string;
+  gender: string;
+  species: string;
+  created: string;
+  status: string;
+  image: string;
+  type?: string;
+  origin?: { name: string; url?: string };
+};
+export type IProps = {
+  card: CardProps;
+  setIsOpen: React.Dispatch<boolean>;
+};
+const Card = ({ setIsOpen, card }: IProps) => {
+  const { id, name, gender, created, species, status, image } = card;
+  const dispatch = useAppDispatch();
+  const handleClick: MouseEventHandler<HTMLDivElement> = (e) => {
+    document.body.classList.add('modalOpen');
+    dispatch(setModalWindow({ id }));
+    setIsOpen(true);
+  };
```

Comment:
very useful article for you, https://habr.com/ru/companies/ruvds/articles/554280/https://habr.com/ru/companies/ruvds/articles/554280/ pay attention on useModalState hook and think how we can separate modal logic in this project